_tools/scalastyle.xml

```diff
@@ -25,122 +25,121 @@
+  <check level="error" class="org.scalastyle.scalariform.IllegalImportsChecker" enabled="true">
+    <parameters>
+      <parameter name="illegalImports"><![CDATA[sun._,java.awt._]]></parameter>
+    </parameters>
+  </check>
+  <check level="error" class="org.scalastyle.scalariform.ParameterNumberChecker" enabled="true">
+    <parameters>
+      <parameter name="maxParameters"><![CDATA[10]]></parameter>
+    </parameters>
+  </check>
+  <check level="error" class="org.scalastyle.scalariform.MagicNumberChecker" enabled="true">
+    <parameters>
+      <parameter name="ignore"><![CDATA[-1,0,1,2,3]]></parameter>
+    </parameters>
+  </check>
+  <check level="error" class="org.scalastyle.scalariform.NoWhitespaceBeforeLeftBracketChecker" enabled="false"/>
+  <check level="error" class="org.scalastyle.scalariform.NoWhitespaceAfterLeftBracketChecker" enabled="false"/>
+  <check level="error" class="org.scalastyle.scalariform.ReturnChecker" enabled="true"/>
+  <!--<check level="error" class="org.scalastyle.scalariform.NullChecker" enabled="true"/>-->
+  <check level="error" class="org.scalastyle.scalariform.NoCloneChecker" enabled="true"/>
+  <check level="error" class="org.scalastyle.scalariform.NoFinalizeChecker" enabled="true"/>
+  <check level="error" class="org.scalastyle.scalariform.CovariantEqualsChecker" enabled="true"/>
+  <check level="error" class="org.scalastyle.scalariform.StructuralTypeChecker" enabled="true"/>
+  <check level="error" class="org.scalastyle.file.RegexChecker" enabled="true">
+    <parameters>
+      <parameter name="regex"><![CDATA[println]]></parameter>
+    </parameters>
+  </check>
+  <check level="error" class="org.scalastyle.scalariform.NumberOfTypesChecker" enabled="true">
+    <parameters>
+      <parameter name="maxTypes"><![CDATA[100]]></parameter>
+    </parameters>
+  </check>
+  <check level="error" class="org.scalastyle.scalariform.NumberOfMethodsInTypeChecker" enabled="true">
+    <parameters>
+      <parameter name="maxMethods"><![CDATA[100]]></parameter>
+    </parameters>
+  </check>
+  <check level="error" class="org.scalastyle.scalariform.CyclomaticComplexityChecker" enabled="true">
+    <parameters>
+      <parameter name="maximum"><![CDATA[25]]></parameter>
+    </parameters>
+  </check>
+  <check level="error" class="org.scalastyle.scalariform.UppercaseLChecker" enabled="true"/>
+  <check level="error" class="org.scalastyle.scalariform.SimplifyBooleanExpressionChecker" enabled="false"/>
+  <check level="error" class="org.scalastyle.scalariform.IfBraceChecker" enabled="true">
+    <parameters>
+      <parameter name="singleLineAllowed"><![CDATA[true]]></parameter>
+      <parameter name="doubleLineAllowed"><![CDATA[true]]></parameter>
+    </parameters>
+  </check>
+  <check level="error" class="org.scalastyle.scalariform.MethodLengthChecker" enabled="true">
+    <parameters>
+      <parameter name="maxLength"><![CDATA[50]]></parameter>
```

Comment:
hi, @felixcheung, thanks for review. This is a crazy pull request.
Keeping the readable code is the purpose, I hate ugly code :-( 

There some relaxed rule, for example, maxLineLength is 120, maxTypes and maxTypes are 100. If more strict, there would be more to modify.
This rule list are checked via `scalatyple:check `